app/src/main/java/android/coding/interview/makeitawesome/networking/RestClient.java

```diff
@@ -0,0 +1,86 @@
+package android.coding.interview.makeitawesome.networking;
+
+import android.coding.interview.makeitawesome.model.Album;
+
+import com.squareup.okhttp.Cache;
+import com.squareup.okhttp.OkHttpClient;
+
+import java.io.File;
+import java.util.List;
+import java.util.UUID;
+import java.util.concurrent.TimeUnit;
+
+import de.halfbit.tinybus.TinyBus;
+import retrofit.Callback;
+import retrofit.RequestInterceptor;
+import retrofit.RestAdapter;
+import retrofit.RetrofitError;
+import retrofit.android.AndroidLog;
+import retrofit.client.OkClient;
+import retrofit.client.Response;
+
+/**
+ * Created by antonio on 24/07/15.
+ */
+public class RestClient {
+
+    private ApiPhoto api;
+
+
+    public static RestClient restClient= null;
+    private static final String URL = "http://jsonplaceholder.typicode.com/photos/";
+
+    protected RestClient(){
+
+        OkHttpClient okHttpClient = new OkHttpClient();
+        File cacheDir = new File(System.getProperty("java.io.tmpdir"), UUID.randomUUID().toString());
+        Cache cache = null;
+        cache = new Cache(cacheDir, 10 * 1024 * 1024);
+
+
+        RequestInterceptor requestInterceptor = new RequestInterceptor() {
+            @Override
+            public void intercept(RequestFacade request) {
+                request.addHeader("content-type", "application/json");
+                request.addHeader("accept-encoding", "gzip");  // Here is the problem
+            }
+        };
+
+        okHttpClient.setCache(cache);
+        okHttpClient.setConnectTimeout(60000, TimeUnit.MILLISECONDS);
+        okHttpClient.setReadTimeout(60000, TimeUnit.MILLISECONDS);
```

Comment:
Well, I guess that adding a cache is kind of an improvement. But the server is explicitly telling not to cache the content of the request (Expires:-1 Pragma:no-cache). So, I don't see the point of the improvement. It's ok as an exercise, but I'd expect some comments explaining that it would be useful in other context.